controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1611,6 +1616,88 @@ bool master_thread::handle_cmdu_1905_topology_notification(const std::string &sr
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_topology_response(const std::string &src_mac,
+                                                       ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received TOPOLOGY_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto tlvDeviceInformation = cmdu_rx.getClass<ieee1905_1::tlvDeviceInformation>();
+    if (!tlvDeviceInformation) {
+        LOG(ERROR) << "ieee1905_1::tlvDeviceInformation not found";
+        return false;
+    }
+
+    const auto &al_mac = tlvDeviceInformation->mac();
+
+    // Get all Agent known fronthaul radios if exist, and compare it to what the Agent reports.
+    // If a radio exist on the database but not the tlvDeviceInformation, this radio node needs
+    // to be removed.
+    // This shall work in the opposite way as well: If it doesn't exist on the database, but
+    // reported in the TLV, then we need to add the node to the database. Of course this will remain
+    // as "TODO" for future task. Meanwhile parse properly the TLV, leaving the unused parts in
+    // comment for future implementation.
+    auto fronthaul_radios_on_db = database.get_node_children(
+        network_utils::mac_to_string(al_mac), beerocks::TYPE_SLAVE, beerocks::STATE_CONNECTED);
+
+    std::unordered_set<std::string> reported_fronthaul_radios;
+
+    for (uint8_t i = 0; i < tlvDeviceInformation->local_interface_list_length(); i++) {
+        const auto iface_info_tuple = tlvDeviceInformation->local_interface_list(i);
+        if (!std::get<0>(iface_info_tuple)) {
+            LOG(ERROR) << "Failed to get " << int(i) << " element of local iface info "
+                       << "on Device Information TLV";
+            return false;
+        }
+
+        auto &iface_info = std::get<1>(iface_info_tuple);
+
+        const auto &iface_mac = iface_info.mac();
+        auto iface_mac_str    = network_utils::mac_to_string(iface_mac);
+
+        // For future implementation
+        // const auto media_type        = iface_info.media_type();
+
+        // For wireless interface it is defined on IEEE 1905.1 that the size of the media info
+        // is n=10 octets, which the size of s802_11SpecificInformation struct.
+        // For wired interface n=0.
+        if (iface_info.media_info_length() == 10) {
```

Comment:
I think you should also check the media type, not just the length. It's possible that with some extension, a different media type with a different media info format happens to have the same length.

Reply:
According to the standard, only wireless interface has a length of 10. But Ok, I added also an enum check.